File: src/main/java/graphql/schema/StaticDataFetcher.java
package graphql.schema;


import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public class StaticDataFetcher implements DataFetcher {


    private final Object value;

    public StaticDataFetcher(Object value) {
        this.value = value;
    }

    @Override
    public CompletionStage<Object> get(DataFetchingEnvironment environment) {
        return CompletableFuture.completedFuture(value);
    }
}

File: src/main/java/graphql/schema/StaticDataFetcher.kt
package graphql.schema


import java.util.concurrent.CompletableFuture
import java.util.concurrent.CompletionStage

class StaticDataFetcher<T>(private val value: T) : DataFetcher<T> {

    override fun get(environment: DataFetchingEnvironment): CompletionStage<T> {
        return CompletableFuture.completedFuture(value)
    }
}
